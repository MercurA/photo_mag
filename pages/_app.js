import '../styles/globals.css'
import { createContext, useReducer } from "react"
import reducer, { initialState } from "../utils/state/reducer"

export const AppContext = createContext(null)

const MyApp = ({ Component, pageProps }) => {
  const [state, dispatch] = useReducer(reducer, initialState)

  return (
    <AppContext.Provider value={{ ...state, dispatch }}>
      <Component {...pageProps} />
    </AppContext.Provider>
  )
}

export default MyApp