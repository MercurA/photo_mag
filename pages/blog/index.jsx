import DesktopBlogContainer from '../../containers/DesktopBlogContainer'

const Blog = () => {
    return (
        <DesktopBlogContainer />
    )
}

export default Blog;