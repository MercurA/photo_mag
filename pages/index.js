import { createContext, useReducer } from "react";
import CardContainer from "../components/card_image/CardContainer";
import Menu from "../components/menu/Menu";
import MusicContainer from "../components/music/MusicContainer";
import reducer, { initialState } from "../utils/state/reducer";

export const AppContext = createContext(null)

export default function Home() {
  const [state, dispatch] = useReducer(reducer, initialState)

  return (
    <AppContext.Provider value={state}>
      <div >
        <Menu dispatch={dispatch}/>
        <CardContainer />
        <MusicContainer dispatch={dispatch}/>
      </div>
    </AppContext.Provider>
  )
}
