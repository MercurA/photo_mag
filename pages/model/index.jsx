import { useRef, useState } from 'react'
import styles from './model.module.css'
import { Canvas, useFrame } from '@react-three/fiber'

const Sphere = ({position, size, color}) => {
    const [hovered, setHovered] = useState(false)
    const sphereRef = useRef()
    useFrame((state, delta) => {
        sphereRef.current.rotation.y += delta/6
    })
    return (
        <mesh ref={sphereRef} onPointerOver={(e) => {e.stopPropagation(); setHovered(!hovered)}}>
            <sphereGeometry />
            <meshStandardMaterial color={hovered ? color : 'lightblue'} wireframe/>
        </mesh>
    )
}

const Model = () => {
    return (
        <div className={styles.container}>
            <Canvas >
                <directionalLight position={[0, 0, 7]} />
                <Sphere position={[0,0,0]} size={[1,1,1]} color={'orange'}/>
            </Canvas>
        </div>
    )
}

export default Model