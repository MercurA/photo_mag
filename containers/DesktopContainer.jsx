import { useContext, useEffect } from "react"

import Galery from "../components/image_galery/Galery"
import Cover from '../components/cover_image/Cover'
import { AppContext } from "../pages"
import { ACTIONS } from "../utils/state/reducer"

const DesktopContainer = () => {
    const context = useContext(AppContext)
    const { dispatch, imageList, currentImageCollection } = context

    useEffect(() => {
        const calcImageToSelect = Math.floor(imageList[currentImageCollection].length / 2)
        dispatch({
            type: ACTIONS.setImageDetails,
            payload: imageList[currentImageCollection][calcImageToSelect],
        })
    }, [currentImageCollection, dispatch, imageList])

    return (
        <>
            <Galery />
            <Cover image={context?.currentImage} />
        </>
    )
}

export default DesktopContainer