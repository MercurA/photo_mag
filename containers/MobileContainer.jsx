import { useContext } from 'react'

import CardContainer from '../components/card_image/CardContainer'
import ImagePortal from '../components/card_image/ImagePortal'
import Menu from "../components/menu/Menu"
import { AppContext } from '../pages/_app'
import Comingsoon from '../components/misc/Comingsoon'

const MobileContainer = () => {
    const context = useContext(AppContext)
    
    const {currentImage, imageList, currentImageCollection} = context

    return (
        <>
            <Menu />
            {imageList[currentImageCollection].length ? <CardContainer /> : <Comingsoon /> }
            {currentImage ? <ImagePortal /> : null}
        </>
    )
}

export default MobileContainer