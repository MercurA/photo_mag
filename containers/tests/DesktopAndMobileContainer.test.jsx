import { render } from "@testing-library/react"
import { AppContext } from "../../pages"
import DesktopContainer from "../DesktopContainer"
import MobileContainer from "../MobileContainer"

describe('Disktop Container', () => {
    let state = {
        dispatch: () => {},
        imageList: {
            'black&white':[ 
                {
                    path: '/images/arh_1.jpg',
                    width: 6016,
                    height: 4016
                },
            ]
        },
        currentImageCollection: 'black&white'
    }
    it('should render Desktop cont correctly', () => {
        const comp = render(
            <AppContext.Provider value={state}>
                <DesktopContainer />
            </AppContext.Provider>
        )
        
        expect(comp).toMatchSnapshot()
    })

    it('should render Desktop cont correctly', () => {
        const comp = render(
            <AppContext.Provider value={state}>
                <MobileContainer />
            </AppContext.Provider>
        )
        
        expect(comp).toMatchSnapshot()
    })
})