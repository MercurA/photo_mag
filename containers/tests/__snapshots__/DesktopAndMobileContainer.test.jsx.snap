// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disktop Container should render Desktop cont correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
        id="galery"
      >
        <div
          class="item"
        >
          <img
            alt=""
            class="image"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=%2Fimages%2Farh_1.jpg&w=3840&q=75"
            srcset="/_next/image?url=%2Fimages%2Farh_1.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="effect"
        >
          <div
            class="commingSoonDesktop"
            id="comingsoon"
          >
            Coming Soon!
          </div>
        </div>
        <div
          class="container"
        >
          <div
            class="brand"
          >
            Horea's Photography
             
          </div>
          <div
            class="categoryGroup"
          >
            <div
              class="text textUnblur"
              data-id="black&white"
              data-index="0"
              id="black&white"
            >
              Black & White
            </div>
            <div
              class="text "
              data-id="landscape"
              data-index="1"
              id="landscape"
            >
              Landscape
            </div>
            <div
              class="text "
              data-id="people&cities"
              data-index="2"
              id="people&cities"
            >
              People and cities
            </div>
          </div>
          <div
            class="contact"
          >
            <p>
              constandacheh@gmail.com
            </p>
            <div>
              <a
                class="link"
                href="https://www.instagram.com/cypher_mercurius/"
                target="_blank"
              >
                <img
                  alt="instagram logo"
                  data-nimg="1"
                  decoding="async"
                  height="20"
                  loading="lazy"
                  src="/icons/insta_white.svg"
                  style="color: transparent;"
                  width="20"
                />
              </a>
              <a
                class="link"
                href="https://github.com/MercurA"
                target="_blank"
              >
                <img
                  alt="github logo"
                  data-nimg="1"
                  decoding="async"
                  height="20"
                  loading="lazy"
                  src="/icons/github_white.svg"
                  style="color: transparent;"
                  width="20"
                />
              </a>
              <a
                class="link"
                href="https://soundcloud.com/dionis5"
                target="_blank"
              >
                <img
                  alt="github logo"
                  data-nimg="1"
                  decoding="async"
                  height="20"
                  loading="lazy"
                  src="/_next/image?url=%2Ficons%2Fsoundcloud.jpg&w=48&q=75"
                  srcset="/_next/image?url=%2Ficons%2Fsoundcloud.jpg&w=32&q=75 1x, /_next/image?url=%2Ficons%2Fsoundcloud.jpg&w=48&q=75 2x"
                  style="color: transparent;"
                  width="20"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
      id="galery"
    >
      <div
        class="item"
      >
        <img
          alt=""
          class="image"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=%2Fimages%2Farh_1.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fimages%2Farh_1.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
    </div>
    <div
      class="container"
    >
      <div
        class="effect"
      >
        <div
          class="commingSoonDesktop"
          id="comingsoon"
        >
          Coming Soon!
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="brand"
        >
          Horea's Photography
           
        </div>
        <div
          class="categoryGroup"
        >
          <div
            class="text textUnblur"
            data-id="black&white"
            data-index="0"
            id="black&white"
          >
            Black & White
          </div>
          <div
            class="text "
            data-id="landscape"
            data-index="1"
            id="landscape"
          >
            Landscape
          </div>
          <div
            class="text "
            data-id="people&cities"
            data-index="2"
            id="people&cities"
          >
            People and cities
          </div>
        </div>
        <div
          class="contact"
        >
          <p>
            constandacheh@gmail.com
          </p>
          <div>
            <a
              class="link"
              href="https://www.instagram.com/cypher_mercurius/"
              target="_blank"
            >
              <img
                alt="instagram logo"
                data-nimg="1"
                decoding="async"
                height="20"
                loading="lazy"
                src="/icons/insta_white.svg"
                style="color: transparent;"
                width="20"
              />
            </a>
            <a
              class="link"
              href="https://github.com/MercurA"
              target="_blank"
            >
              <img
                alt="github logo"
                data-nimg="1"
                decoding="async"
                height="20"
                loading="lazy"
                src="/icons/github_white.svg"
                style="color: transparent;"
                width="20"
              />
            </a>
            <a
              class="link"
              href="https://soundcloud.com/dionis5"
              target="_blank"
            >
              <img
                alt="github logo"
                data-nimg="1"
                decoding="async"
                height="20"
                loading="lazy"
                src="/_next/image?url=%2Ficons%2Fsoundcloud.jpg&w=48&q=75"
                srcset="/_next/image?url=%2Ficons%2Fsoundcloud.jpg&w=32&q=75 1x, /_next/image?url=%2Ficons%2Fsoundcloud.jpg&w=48&q=75 2x"
                style="color: transparent;"
                width="20"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Disktop Container should render Desktop cont correctly 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="menuContainer undefined"
      >
        <div
          class="menuWrapper"
        >
          <div
            class="slideMenuBtn"
          >
            <span />
            <span />
            <span />
          </div>
          <div
            class="brandTitle"
          >
            <strong>
              Horea's
            </strong>
             Photography
          </div>
        </div>
        <div
          class="drawer drawerClosed"
        >
          <div
            class="drawerHeader"
          >
            <button
              class="drawerCloseBtn"
            >
              <img
                alt="close icon"
                data-nimg="1"
                decoding="async"
                height="15"
                loading="lazy"
                src="/icons/close.svg"
                style="color: transparent;"
                width="15"
              />
            </button>
          </div>
          <div
            class="drawerBody"
          >
            <div
              class="drawerItem"
              data-id="black&white"
            >
              Black & White
            </div>
            <div
              class="drawerItem"
              data-id="landscape"
            >
              Landscape
            </div>
            <div
              class="drawerItem"
              data-id="people&cities"
            >
              People and cities
            </div>
          </div>
          <div
            class="drawerFooter"
          >
            <p>
              constandacheh@gmail.com
            </p>
            <div>
              <a
                class="link"
                href="https://www.instagram.com/cypher_mercurius/"
                target="_blank"
              >
                <img
                  alt="instagram logo"
                  data-nimg="1"
                  decoding="async"
                  height="20"
                  loading="lazy"
                  src="/icons/insta_white.svg"
                  style="color: transparent;"
                  width="20"
                />
              </a>
              <a
                class="link"
                href="https://github.com/MercurA"
                target="_blank"
              >
                <img
                  alt="github logo"
                  data-nimg="1"
                  decoding="async"
                  height="20"
                  loading="lazy"
                  src="/icons/github_white.svg"
                  style="color: transparent;"
                  width="20"
                />
              </a>
              <a
                class="link"
                href="https://soundcloud.com/dionis5"
                target="_blank"
              >
                <img
                  alt="github logo"
                  data-nimg="1"
                  decoding="async"
                  height="20"
                  loading="lazy"
                  src="/_next/image?url=%2Ficons%2Fsoundcloud.jpg&w=48&q=75"
                  srcset="/_next/image?url=%2Ficons%2Fsoundcloud.jpg&w=32&q=75 1x, /_next/image?url=%2Ficons%2Fsoundcloud.jpg&w=48&q=75 2x"
                  style="color: transparent;"
                  width="20"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="cardWrapper"
        >
          <img
            alt=""
            class="image"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=%2Fimages%2Farh_1.jpg&w=3840&q=75"
            srcset="/_next/image?url=%2Fimages%2Farh_1.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="menuContainer undefined"
    >
      <div
        class="menuWrapper"
      >
        <div
          class="slideMenuBtn"
        >
          <span />
          <span />
          <span />
        </div>
        <div
          class="brandTitle"
        >
          <strong>
            Horea's
          </strong>
           Photography
        </div>
      </div>
      <div
        class="drawer drawerClosed"
      >
        <div
          class="drawerHeader"
        >
          <button
            class="drawerCloseBtn"
          >
            <img
              alt="close icon"
              data-nimg="1"
              decoding="async"
              height="15"
              loading="lazy"
              src="/icons/close.svg"
              style="color: transparent;"
              width="15"
            />
          </button>
        </div>
        <div
          class="drawerBody"
        >
          <div
            class="drawerItem"
            data-id="black&white"
          >
            Black & White
          </div>
          <div
            class="drawerItem"
            data-id="landscape"
          >
            Landscape
          </div>
          <div
            class="drawerItem"
            data-id="people&cities"
          >
            People and cities
          </div>
        </div>
        <div
          class="drawerFooter"
        >
          <p>
            constandacheh@gmail.com
          </p>
          <div>
            <a
              class="link"
              href="https://www.instagram.com/cypher_mercurius/"
              target="_blank"
            >
              <img
                alt="instagram logo"
                data-nimg="1"
                decoding="async"
                height="20"
                loading="lazy"
                src="/icons/insta_white.svg"
                style="color: transparent;"
                width="20"
              />
            </a>
            <a
              class="link"
              href="https://github.com/MercurA"
              target="_blank"
            >
              <img
                alt="github logo"
                data-nimg="1"
                decoding="async"
                height="20"
                loading="lazy"
                src="/icons/github_white.svg"
                style="color: transparent;"
                width="20"
              />
            </a>
            <a
              class="link"
              href="https://soundcloud.com/dionis5"
              target="_blank"
            >
              <img
                alt="github logo"
                data-nimg="1"
                decoding="async"
                height="20"
                loading="lazy"
                src="/_next/image?url=%2Ficons%2Fsoundcloud.jpg&w=48&q=75"
                srcset="/_next/image?url=%2Ficons%2Fsoundcloud.jpg&w=32&q=75 1x, /_next/image?url=%2Ficons%2Fsoundcloud.jpg&w=48&q=75 2x"
                style="color: transparent;"
                width="20"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="container"
    >
      <div
        class="cardWrapper"
      >
        <img
          alt=""
          class="image"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=%2Fimages%2Farh_1.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fimages%2Farh_1.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Farh_1.jpg&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
